src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -9571,6 +9571,9 @@ void visitInterpolatedStringHandlerConstructor()
                     return;
                 }
 
+                // In new extension form, the nullable rewriter processes the arguments as if receiver is the first item in the argument list, like the old extension form. This means that all of
+                // our placeholders will be off-by-one, with the extension receiver in the first position.
+                var newExtensionFormOffset = handlerData.ArgumentPlaceholders.Any(p => p.ArgumentIndex == BoundInterpolatedStringArgumentPlaceholder.ExtensionReceiver) ? 1 : 0;
```

Comment:
>BoundInterpolatedStringArgumentPlaceholder.ExtensionReceiver [](http://example.com/codeflow?start=106&length=60)

This logic doesn't look right. I think the adjustment should happen even if receiver is not referenced. If an instance extension member is invoked, we should adjust indexes.